### PR TITLE
Upgrade use of semconv to v1.24.0

### PR DESCRIPTION
**Pull request by pellared**
Fixes https://github.com/open-telemetry/opentelemetry-go/issues/4753

Related to:
- https://github.com/open-telemetry/opentelemetry-specification/pull/3794

### CODE REVIEW

File: CHANGELOG.md
Comment (pellared) [inline CHANGELOG.md:25]: **Notice**

These changes follows https://github.com/open-telemetry/opentelemetry-specification/pull/3713 (released in https://github.com/open-telemetry/semantic-conventions/releases/tag/v1.22.0) which are not backwards compatible.

The problem is that updated mappings will work great only for code that is uses semver >= v1.22.0. 

If an instrumentation library is using e.g. v1.20.0 semver schema then it may now miss the mapping. E.g.  `otelhttp`, `otelgrpc` add `net.sock.peer.addr` attribute and these instrumentation libraries will no longer nicely integrate with the exporter.

Brainstorming: We could probably handle some of changes like this in a backwards compatible way if we would do the schema translation for the spans that are coming to the exporter. But this would require https://github.com/orgs/open-telemetry/projects/64.

Comment (pellared) [reply CHANGELOG.md:25]: People can always use previous version of the exporter if they do not like the translation changes.

Comment (pellared) [reply CHANGELOG.md:25]: SIG meeting: We can have a fallback to handle previous semconv.

Comment (pellared) [reply CHANGELOG.md:25]: I found more breaking changes.
I created https://github.com/open-telemetry/opentelemetry-specification/issues/3793 and https://github.com/open-telemetry/opentelemetry-specification/pull/3794.

Marking as blocked by specification.

Comment (pellared) [reply CHANGELOG.md]: I got 5 approvals in https://github.com/open-telemetry/opentelemetry-specification/pull/3794 so far so I decided to implement as I proposed.

File: CHANGELOG.md
Comment (MrAlias) [inline CHANGELOG.md:27]:
```suggestion
- Upgrade all use of `go.opentelemetry.io/otel/semconv` to use `v1.24.0`. (#4754)
```

**Review by MrAlias — APPROVED** [review]

File: CHANGELOG.md
Comment (pellared) [reply CHANGELOG.md]: The PR is merged. We can wait for a spec release, but I think it is safe to review and merge (if approved).

**Review by hanyuancheung — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Comment by pellared** [comment]
I plan to merge this tomorrow.

**Review by MadVikingGod — APPROVED** [review]

File: CHANGELOG.md
Comment (jufemaiz) [inline CHANGELOG.md:29]: Looks a lot more like it's using v1.24.0. May lead to confusion?

Comment (pellared) [reply CHANGELOG.md:29]: You are correct. Thanks ❤️

**Pull request merged**
